channel_dns.py

```diff
@@ -161,12 +162,12 @@ def look_for_source_data(self, token=None, value=None):
             linux_inotify        = re.compile('([A-Za-z0-9.-]*)\.L[0-9]{2}\.', re.IGNORECASE)
             dtrace_process       = re.compile('([0-9]+)\.([A-Za-z0-9-=]+)\.h\.([A-Za-z0-9.-=]+)\.c\.([A-Za-z0-9.-=]+)\.D1\.', re.IGNORECASE)
             dtrace_file_open     = re.compile('([0-9]+)\.([A-Za-z0-9-=]+)\.h\.([A-Za-z0-9.-=]+)\.f\.([A-Za-z0-9.-=]+)\.D2\.', re.IGNORECASE)
-            desktop_ini_browsing = re.compile('([^\.]+)\.([^\.]+)\.ini\.', re.IGNORECASE)
+            desktop_ini_browsing = re.compile('([^\.]+)\.([^\.]+)\.([^\.]+)\.ini\.', re.IGNORECASE)
             aws_keys_event       = re.compile('([A-Za-z0-9-]*)\.([A-Za-z0-9.-]*)\.A[0-9]{3}\.', re.IGNORECASE)
 
             m = desktop_ini_browsing.match(value)
             if m:
-                return self._desktop_ini_browsing(username=m.group(1), domain=m.group(2))
+                return self._desktop_ini_browsing(username=m.group(1), hostname=m.group(2), domain=m.group(3))
```

Comment:
Could we change the hostname and domain around here please. i.e. `domain=m.group(2), host=m.group(3)`.

My reasoning here is backward compat. We will still potentially have older tokens that when triggered will only send through the domain and username. With this in mind we will need to make sure other paths are affected too. 

I can not remember off the top of head, but you may need to check that `m.group` has 3 items before trying to assign it to `hostname`. You may need another `if statement checking the number of `m.group` items then making the appropriate call.

Reply:
`len()` checker built-in. Good call on backwards compatibility.